### PR TITLE
[ENVIRONMENT] Add support for providing the deployer environment as parameter

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
The are cases when the deployer is in a different environment from the sap library 

## Solution
Add the support for multiple environment in the naming module, i.e deployer environment and sdu environment. Rename the vnet variable inside the naming module to landscape as it better reflects the design

## Tests
<Please provide steps to test the PR>

## Notes
<Additional comments for the PR>

### CODE REVIEW

**Comment by YunziZhangAzure** [comment]
Please resolve the conflicts. Also, I notice some codes in this PR also exists in another PR #948 , maybe refactor that?

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 1]
```diff
@@ -31,14 +31,14 @@ locals {
...
-  vnet_mgmt_name   = local.vnet_mgmt_exists ? split("/", local.vnet_mgmt_arm_id)[8] : try(local.vnet_mgmt.name, format("%s-vnet", local.prefix))
+  vnet_mgmt_name   = local.vnet_mgmt_exists ? split("/", local.vnet_mgmt_arm_id)[8] : try(local.vnet_mgmt.name, format("%s%s", local.prefix,local.resource_suffixes.vnet))
   vnet_mgmt_addr   = local.vnet_mgmt_exists ? "" : try(local.vnet_mgmt.address_space, "")
 
   // Management subnet
   sub_mgmt          = try(local.vnet_mgmt.subnet_mgmt, {})
   sub_mgmt_arm_id   = try(local.sub_mgmt.arm_id, "")
   sub_mgmt_exists   = length(local.sub_mgmt_arm_id) > 0 ? true : false
-  sub_mgmt_name     = local.sub_mgmt_exists ? split("/", local.sub_mgmt_arm_id)[10] : try(local.sub_mgmt.name, format("%s_deployment-subnet", local.prefix))
+  sub_mgmt_name     = local.sub_mgmt_exists ? split("/", local.sub_mgmt_arm_id)[10] : try(local.sub_mgmt.name, format("%s%s", local.prefix,,local.resource_suffixes.deployer_subnet))
```
Comment (YunziZhangAzure) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:41]: local.prefix,,local.resource_suffixes.deployer_subnet

a redundant comma?

**Review by YunziZhangAzure — COMMENTED** [review]
There are still some codes which already exist in feature/remote-tfstate2, thus, they are not supposed to show in this PR. Could you please refactor them? only put new changes in this PR. If it is hard to rebase, maybe consider starting a new PR based on current feature/remote-tfstate2 branch.

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf (hunk 1 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:41]: fixed

File: deploy/terraform/terraform-units/modules/sap_namegenerator/resourcegroup.tf [hunk 2]
```diff
@@ -14,6 +14,6 @@ locals {
   landscape_storageaccount_name      = substr(replace(lower(format("%s%s%sdiag%s", local.landscape_env_verified, local.location_short, local.sap_vnet_verified, local.random_id_verified)), "/[^a-z0-9]/", ""), 0, var.azlimits.stgaccnt)
   library_storageaccount_name        = substr(replace(lower(format("%s%ssaplib%s", local.library_env_verified, local.location_short, local.random_id_verified)), "/[^a-z0-9]/", ""), 0, var.azlimits.stgaccnt)
   sdu_storageaccount_name            = substr(replace(lower(format("%s%s%sdiag%s", local.env_verified, local.location_short, local.sap_vnet_verified, local.random_id_verified)), "/[^a-z0-9]/", ""), 0, var.azlimits.stgaccnt)
-  terraformstate_storageaccount_name = substr(replace(lower(format("%s%stfstate%s", local.library_env_verified, local.location_short, local.random_id_verified)), "/[^a-z0-9]/", ""), 0, var.azlimits.stgaccnt)
+  terraformstate_storageaccount_name = substr(replace(lower(format("%s%s%stfstate%s", local.library_env_verified, local.location_short, local.sap_vnet_verified, local.random_id_verified)), "/[^a-z0-9]/", ""), 0, var.azlimits.stgaccnt)
```
Comment (YunziZhangAzure) [inline deploy/terraform/terraform-units/modules/sap_namegenerator/resourcegroup.tf:17]: I check the naming standard about storage account. Unfortunately, it only describes the storage account of saplibrary, but I guess storage account of tfstate is similar.
{environment(5char)}{region_map(4CHAR)}saplib(12CHAR){RND(3CHAR)}
Ex: protowus2saplibxxx

It seems sap_vnet is not part of its name.

**Review by YunziZhangAzure — APPROVED** [review]

**Pull request merged**